mlir/lib/Target/LLVM/CMakeLists.txt

```diff
@@ -209,3 +209,35 @@ if(MLIR_ENABLE_ROCM_CONVERSIONS)
   )
 endif()
 
+if ("SPIRV" IN_LIST LLVM_TARGETS_TO_BUILD)
+  set(SPIRV_LIBS
+    SPIRVCodeGen
+
+  )
+endif()
+
+add_mlir_dialect_library(MLIRXeVMTarget
+  XeVM/Target.cpp
+
+  OBJECT
+
+  ADDITIONAL_HEADER_DIRS
+  ${MLIR_MAIN_INCLUDE_DIR}/mlir/Dialect/LLVMIR
+
+  LINK_COMPONENTS
+  ${SPIRV_LIBS}
+
+  LINK_LIBS PUBLIC
+  MLIRIR
+  MLIRExecutionEngineUtils
```

Comment:
do we need to depend on ExeuctionEngine and GPUDialect?

Reply:
XeVM target needs to access things like  gpu::TargetAttrInterface and gpu::TargetOptions

Reply:
And base ModuleToObject requires makeOptimizingTransformer from ExecutionEngine